server/events/project_command_runner.go

```diff
@@ -313,11 +314,25 @@ func (p *DefaultProjectCommandRunner) doPlan(ctx models.ProjectCommandContext) (
 		return nil, "", DirNotExistErr{RepoRelDir: ctx.RepoRelDir}
 	}
 
+	lockFileName := ".terraform.lock.hcl"
+	lockfilePath := filepath.Join(projAbsPath, lockFileName)
+	lockFileExistsProirToPlan := common.FileExists(lockfilePath)
+
 	outputs, err := p.runSteps(ctx.Steps, ctx, projAbsPath)
 	if err != nil {
 		if unlockErr := lockAttempt.UnlockFn(); unlockErr != nil {
 			ctx.Log.Err("error unlocking state after plan error: %v", unlockErr)
 		}
+		if !lockFileExistsProirToPlan && common.FileExists(lockfilePath) {
+			// if the lock file did not exist before running the plan delete the lock file that was generated
```

Comment:
Why is this only being run when there is an error in the steps?  Does this file need to be present for an apply?  Maybe it might make sense to use something like git to detect whether this is a new file that's been generated previously and clean this up before the init runs in the step runner itself.

Reply:
Good spot I have moved this outside the `if err != nil {`